invenio_drafts_resources/services/records/components/files_base.py

```diff
@@ -0,0 +1,239 @@
+            errors.append(
+                {
+                    "field": f"f{self.files_data_key}.default_preview",
+                    "messages": e.messages,
+                }
+            )
+
+    def edit(self, identity, draft=None, record=None):
+        """Edit callback."""
+        draft_files = self.get_record_files(draft)
+        record_files = self.get_record_files(record)
+        draft_bucket = self.get_record_bucket(draft)
+        if draft_bucket is None:
+            # Happens, when a soft-deleted draft is un-deleted.
+            draft[self.files_data_key] = {"enabled": True}
+            # re-fetch the files attribute - above data getter sets the attribute
+            draft_files = self.get_record_files(draft)
+            draft_files.create_bucket()
+        draft_files.copy(record_files)
+        # force going through the new version
+        draft_files.lock()
+
+    def new_version(self, identity, draft=None, record=None):
+        """New version callback."""
+        # We don't copy files from the previous version, but instead allow
+        # users to import the files.
+        draft_files = self.get_record_files(draft)
+        record_files = self.get_record_files(record)
+        draft_files.enabled = record_files.enabled
+
+    def _publish_new(self, identity, draft, record):
+        """Action when publishing a new draft."""
+        # For unpublished drafts (new and new version), we move the draft
+        # bucket from the draft to the record (instead of creating a new, and
+        # deleting one). For consistency, we keep a bucket for all records
+        # independently of if they have files enabled or not.
+        record_files = self.get_record_files(record)
+        draft_files = self.get_record_files(draft)
+        draft_bucket = self.get_record_bucket(draft)
+        record_files.set_bucket(draft_bucket)
+        record_files.copy(draft_files, copy_obj=False)
+
+        # Lock the bucket
+        # TODO: Make the locking step optional in the future (so
+        # instances can potentially allow files changes if desired).
+        record_files.lock()
+
+        # Cleanup
+        if draft_files.enabled:
+            draft_files.delete_all(remove_obj=False)
+        draft_files.unset_bucket()
+
+    def _publish_edit(self, identity, draft, record):
+        """Action when publishing an edit to an existing record."""
+        # TODO: For published records, we should sync changes from the
+        # draft bucket to the record bucket, so that an instance could
+        # potentially allow a user to update files. For now, sync() only
+        # changes the default_preview and order
+        record_files = self.get_record_files(record)
+        draft_files = self.get_record_files(draft)
+
+        record_files.sync(draft_files)
+
+        # Teardown the bucket and files created in edit().
+        if draft_files.enabled:
+            draft_files.delete_all()
+        draft_files.unlock()
```

Comment:
addition of unlocking the bucket, because of https://github.com/inveniosoftware/invenio-drafts-resources/pull/231/files#r1247649625